.github/workflows/ci-pipeline.yml

```diff
@@ -0,0 +1,78 @@
+name: CI Pipeline for DoDo
+
+on:
+  pull_request:
+    branches:
+      - develop
+      - main
+  push:
+    branches:
+      - develop
+
+jobs:
+  test_and_build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v3
+
+    - name: Set up Docker Compose
+      run: |
+        docker compose version
+
+    - name: Start Services
+      run: |
+        docker compose up -d --build
+      env:
+        POSTGRES_DB: dodo
+        POSTGRES_USER: dodo_user
+        POSTGRES_PASSWORD: dodo_password
+
+    - name: Wait for PostgreSQL to be ready
+      run: |
+        for i in {1..10}; do
+          docker compose exec db pg_isready -U dodo_user -d dodo && break
+          sleep 5
+        done
+
+    - name: Run Migrations
+      run: |
+        docker compose exec web python manage.py migrate
+
+    - name: Run Tests
+      run: |
+        docker compose exec web python manage.py test
+      continue-on-error: false
+
+    - name: Send Slack Failure Notification
+      if: failure()
+      env:
+        SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
+      run: |
+        curl -X POST -H 'Content-type: application/json' --data '{"text":"Pipeline Failed: Tests failed."}' $SLACK_WEBHOOK_URL
+
+    - name: Build Docker Image
+      if: success()
+      run: |
+        docker build -t touqeerjadoon55/dodo:latest .
+
+    - name: Login to DockerHub
+      if: success()
+      env:
+        DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+        DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
+      run: |
+        echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin
+
+    - name: Push Docker Image
+      if: success()
+      run: |
+        docker push touqeerjadoon55/dodo:latest
+
```

Comment:
Do you think we should also remove the image after pushing to dockerhub?

Reply:
Removing the image after pushing prevents unnecessary disk space usage in the CI runner. I will add the docker image remove step.